magento/magento2
# CheckoutAgreements getList refactoring

### Description
Introduced new listing interface for Checkout Agreements with ability to set search criteria.

### Manual testing scenarios
1. Covered with api-functional tests
2. Verify main cases with agreements on checkout flow 

### Contribution checklist
